### PR TITLE
School library basic UI

### PR DESCRIPTION
**Pull request by iankiema**
The following files have been added for the basic UI:
-App.rb
-Main.rb

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    puts 'Enter Specialization:'
+    specialization_label = gets.chomp
+
+    specialization = find_or_create_specialization(specialization_label)
+
+    teacher = Teacher.new(name: name, age: age, specialization: specialization, parent_permission: parent_permission)
+    @people << teacher
+    puts "Teacher #{name} created with ID #{teacher.id}"
+  end
+
```
Comment (KanzaTahreem) [inline app.rb]: - [ ] Great effort so far 👌👏 While creating a teacher, we are not expected to ask for parent permission according to the example video we have ✨ Please make sure that the school library console app behaves in the same was as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) 💪🙌

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  def list_people
+    puts 'List of People:'
+    @people.each { |person| puts "Name:#{person.name} ID: #{person.id}  Age:#{person.age}" }
+  end
+
+  def create_person
+    puts 'Chosse the type of person to create:'
+    puts '1. Teacher'
+    puts '2. Student'
+    type_option = gets.chomp.to_i
```
Comment (KanzaTahreem) [inline app.rb]: - [ ] While listing people, please make sure that we have their respective role as well as we have in the given [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ). Right now, it is not clear weather a person is a STUDENT or a TEACHER. Could you please fix this? 💪

Current | Expected 
---| ---|
image | image

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    book = @books.find { |b| b.title == book_title }
+
+    if person && book
+      rental = Rental.new(Date.today, book, person)
+      @rentals << rental
+      puts "Rental created for #{person.name} (ID: #{person.id}) - #{book.title}"
+    else
+      puts 'Person or book not found'
+    end
+  end
```
Comment (KanzaTahreem) [inline app.rb]: - [ ] Great work so far @iankiema 👏✨ However, the create_rental method is not working according to the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) we have.  According to the example video, while creating a rental, we'll be first presented with the list of the available books and then available people. And then we need to add the date ✨ Also we are not expected to enter the ID of the person, rather we need the list number. However, the current implementation is quite different. Kindly make sure that the app behaves in the same was as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) provided


Current | Expected
---| ---|
image | image

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    if person
+      puts "Rentals for #{person.name} (ID: #{person.id}):"
+      person.rentals.each do |rental|
+        puts "#{rental.book.title} - #{rental.Date}"
+      end
+    else
+      puts 'Person not found.'
+    end
+  end
+end
```
Comment (KanzaTahreem) [inline app.rb]: - [ ] Great work so far 👏👏 However, I noticed that I am unable to list the rentals that were added. Kindly have a look at the following image for better clarity👇 Kindly make sure to fix this and ensure that it is displayed according to the example video we have

image

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @iankiema 

Good job so far👏! I really appreciate the effort you put in!  **_However, there are few things that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Good job with the console app
- ✅ Descriptive README file
- ✅ Linters are working

## Required Changes 🛑

_Please check the comments under the review._

## Optional suggestions ❕

N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    welcome_message
+    display_menu
+
+    choice = gets.chomp.to_i
+
+    process_choice(choice, app)
+
+    break if choice == 7
+  end
+end
```
Comment (danmainah) [inline main.rb]: - [ ] [optional] @iankiema Great work implementing the changes requested, I would also recommend you consider ensuring the welcome message is only when the user opens the app and not when the user selects options in the menu.

**Review by danmainah — APPROVED** [review]
Hi @iankiema , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the changes requested to ensure the UI behavior matches with the given example.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 5 above)
Comment (iankiema) [reply main.rb]: @danmainah thank you for the approval and also the optional suggestion. I have corrected it.Cheers.

**Pull request merged**